src/mlpack/tests/adaboost_regressor_test.cpp

```diff
@@ -0,0 +1,201 @@
+ * when the elements are all equal.
+ */
+TEST_CASE("ExponentialLossEqualTest", "[AdaBoostRegressorTest]")
+{
+  arma::rowvec values(10);
+  values.fill(5);
+  arma::rowvec loss = ExponentialLoss::Calculate(values);
+  for(size_t i = 0; i < 10; i++)
+    REQUIRE(loss[i] == Approx(0.63212).margin(1e-5));
+}
+
+/**
+ * Making sure the LinearLoss is evaluated correctly by doing calculation by
+ * hand
+ */
+TEST_CASE("LinearLossHandCalculation", "[AdaBoostRegressorTest]")
+{
+  arma::rowvec values = {4., 2., 3., 4., 13., 6., 20., 8., 9., 10.};
+  
+  // Hand calculated loss.
+  arma::rowvec loss = {0.2, 0.1, 0.15, 0.2, 0.65, 0.3, 1. , 0.4, 0.45, 0.5};
+
+  arma::rowvec result = LinearLoss::Calculate(values);
+  for(size_t i = 0; i < 10; i++)
+    REQUIRE(result[i] == Approx(loss[i]).margin(1e-5));
+}
+
+/**
+ * Making sure the SquareLoss is evaluated correctly by doing calculation by
+ * hand
+ */
+TEST_CASE("SquareLossHandCalculation", "[AdaBoostRegressorTest]")
+{
+  arma::rowvec values = {4., 2., 3., 4., 13., 6., 20., 8., 9., 10.};
+  
+  // Hand calculated loss.
+  arma::rowvec loss = {0.04, 0.01, 0.0225, 0.04, 0.4225, 0.09, 1., 0.16, 
+                       0.2025, 0.25};
+
+  arma::rowvec result = SquareLoss::Calculate(values);
+  for(size_t i = 0; i < 10; i++)
+    REQUIRE(result[i] == Approx(loss[i]).margin(1e-5));
+}
+
+/**
+ * Making sure the ExponentialLoss is evaluated correctly by doing calculation by
+ * hand
+ */
+TEST_CASE("ExponentialLossHandCalculation", "[AdaBoostRegressorTest]")
+{
+  arma::rowvec values = {4., 2., 3., 4., 13., 6., 20., 8., 9., 10.};
+  
+  // Hand calculated loss(rounded-off).
+  arma::rowvec loss = {0.18126, 0.09516, 0.13929, 0.18126, 0.47795, 
+                       0.25918, 0.63212, 0.32967, 0.36237, 0.39346};
+
+  arma::rowvec result = ExponentialLoss::Calculate(values);
+  for(size_t i = 0; i < 10; i++)
+    REQUIRE(loss[i] == Approx(result[i]).margin(1e-5));
+}
+
+/**
+ * Test that the model is able to perfectly fit all the obvious splits present
+ * in the data and only use 1 tree.
+ *
+ *     |
+ *     |
+ *   2 |            xxxxxx
+ *     |
+ *     |
+ *   1 |      xxxxxx      xxxxxx
+ *     |
+ *     |
+ *   0 |xxxxxx                  xxxxxx
+ *     |___________________________________
+ */
+TEST_CASE("EarlyTerminateTest", "[AdaBoostRegressorTest]")
+{
+  arma::mat dataset;
+  arma::rowvec responses;
+  arma::rowvec values = {0.0, 1.0, 2.0, 1.0, 0.0};
+
+  CreateMultipleSplitData(dataset, responses, 1000, values);
+
+  AdaBoostRegressor<> abr(dataset, responses, 20 /*numTrees*/, 1/*minLeaves*/, 
+                        0/*minGainSplit*/, 0/*maxDepth*/);
+  arma::rowvec preds;
+  abr.Predict(dataset, preds);
+
+  // Ensure that the predictions are perfect.
+  for (size_t i = 0; i < responses.n_elem; ++i)
+    REQUIRE(preds[i] == responses[i]);
+
+  // Ensure that number of trees trained is 1, since 1st tree can do perfect 
+  // predictions.
+  REQUIRE(abr.NumTrees() == 1);
+}
+
```

Comment:
Could you add another test with arma::fmat to use if using floating points gives the same results as double?